### PR TITLE
Terminals for Pods (like kubectl exec)...

### PR DESCRIPTION
**Pull request by kans**
todo:

- [x] life cycle code cleanup
- [x] show loading/error status
- ~[ ] expose a connect command?~ <- too much for now
- ~[ ] do stuff when we lose the pod or WS?~ <- would need to interact with a higher component :-/
- [x] lazy load

### CODE REVIEW

File: frontend/public/components/pod-exec.jsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      ns: metadata.namespace,
+      name: metadata.name,
+      path: 'exec',
+      queryParams: {
+        stdout: 1,
+        stdin: 1,
+        stderr: 1,
+        tty: 1,
+        container: activeContainer,
+        command: '%2Fbin%2Fsh&command=-i&command=-c&command=TERM%3Dxterm%20%2Fbin%2Fsh',
```
Comment (ggreer) [inline frontend/public/components/pod-exec.jsx]: It'd be nice to have the decoded command here. Something like...

```javascript
const command = ['/bin/sh', '-i', '-c', 'TERM=xterm', '/bin/sh'];
...
queryParams: {
  command: command.map(c => encodeURIComponent(c)).join('&command='),
...
```

File: frontend/public/components/pod-exec.jsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    this.ws = new WSFactory(`${metadata.name}-terminal`, {
+      host: 'auto',
+      reconnect: true,
+      path: resourceURL(PodModel, params),
+      jsonParse: false,
+      subProtocols: ['base64.channel.k8s.io'],
+    })
+      .onmessage(raw => {
+        // const fd = raw[0];
```
Comment (ggreer) [inline frontend/public/components/pod-exec.jsx]: Best to delete this and add a comment explaining the reason for the slice below.

File: frontend/public/components/terminal.jsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      // tell the terminal to resize itself
+      terminal.fit();
+      // update the pty
+      this.props.onResize(terminal.rows, terminal.cols);
+    });
+  }
+
+  render () {
+    return <div ref={this.outerRef} style={this.state} className={this.props.className}>
+      <div key={this.terminal} ref={this.innerRef} className="console" />
```
Comment (ggreer) [inline frontend/public/components/terminal.jsx]: Why do we need a key here?

File: frontend/public/components/pod-exec.jsx (hunk 1 above)
Comment (kans) [reply frontend/public/components/pod-exec.jsx]: yes, that would make sense.

File: frontend/public/components/terminal.jsx (hunk 3 above)
Comment (kans) [reply frontend/public/components/terminal.jsx]: I'm not sure we do or even if keys have any meaning outside of arrays- it was copy/pasted.

**Pull request merged**